content/blog/2019-11-mio-v0.7-alpha.md

```diff
@@ -0,0 +1,201 @@
+platforms.
+
+Users that previously used one-shot triggers should now `deregister` the I/O
+source if they no longer want to receive events for it. And once more events are
+wanted the resource should be `register`ed again.
+
+Users that used level triggers basically need to put a loop around all I/O
+operations. When using edge triggers the user is responsible for attempting the
+I/O operation, e.g. a `read`, until it returns a would block error (an
+`io::Error` of kind `io::ErrorKind::WouldBlock`). Only once an operation
+returned a would block error the source will receive more events for that kind
+of operation. Below is a sample example of how to read from a `TcpStream` using
+edge triggers.
+
+```rust
+let stream: TcpStream = ...;
+
+// After we polled we received an readable event for our `TcpStream`.
+
+let mut buf = [0; 4096];
+// With edge triggers we need to read all data available on the socket.
+loop {
+    let bytes_read = match stream.read(&mut buf) {
+        // Read successful, proceed like normal.
+        Ok(bytes_read) => bytes_read,
+        // No more data to read at the moment. Now we will receive another event
+        // once more data is available to read.
+        Err(ref err) if err.kind() == io::ErrorKind::WouldBlock => break,
+        // Got interrupted, try again.
+        Err(ref err) if err.kind() == io::ErrorKind::Interrupted => continue,
+        // Hit an actual error.
+        Err(err) => return Err(err),
+    };
+
+    process_byte(&buf[0..bytes_read]);
+}
+```
+
+**Note**: this needs to happen for all I/O operations on registered event source
+so be careful when using automated tools to convert to the new register API.
+
+
+### Addition of Unix socket API
+
+New in the `net` module are the `UnixListener`, `UnixStream` and `UnixDatagram`
+types that support Unix sockets with an API similar to that found in the
+standard library.
+
+These APIs are currently only support on Unix based OSes.
+
+
+### Removal of deprecated API
+
+In Mio version 0.6 various types, such as the old event loop related and channel
+types, were deprecated. In version 0.7 all deprecated type were removed.
+
+
+### Removed support for OSes
+
+Support for the following OSes is dropped:
+
+ * Linux below version 4, we using newer APIs not present on old Linux versions.
+ * Fuchsia, didn't have any CI coverage and we don't have enough maintainers to
+   properly support it.
+ * Bitrig, development on the OS seems to have stopped and rustc no longer
+   supports it.
+
+<!-- TODO: refer to a document that specifies supported OS and versions? -->
+
+
+### Increased minimum Rust version
+
+As of writing this the minimum Rust version is increased to 1.36, for version
+1.0 we aimed to target the same Rustc version in which async becomes stable (as
+this is major feature for which a lot of crates that dependent on this crate
+have been waiting). Again at the time writing this seems to be Rust 1.39, so
```

Comment:
I suggest

As of this writing that would be Rust 1.39.0, so keep those compilers up to date!